packages/control-group/src/index.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import {ButtonAppearanceType} from '@smashing/button'
+import {Radio, RadioAppearanceType} from '@smashing/radio'
+import {Checkbox, CheckboxAppearanceType} from '@smashing/checkbox'
+import {
+  ControlProps,
+  ControlGroupProps,
+  ControlAppearanceType,
+  ControlGroupAppearanceType
+} from './types'
+import {useDefaults, safeInvoke} from '@smashing/theme'
+import {Strong} from '@smashing/typography'
+import {
+  ControlButton,
+  ControlGroupWrapper,
+  MoreButton,
+  MoreButtonArrow
+} from './styles'
+
+// TODO: add icon component
+const ArrowIcon = () => (
+  <svg
+    width="16"
+    height="10"
+    viewBox="0 0 16 10"
+    fill="#1D304E"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M0.310946 0.808831C0.725541 0.397056 1.39773 0.397056 1.81233 0.808831L8 6.95442L14.1877 0.808831C14.6023 0.397056 15.2745 0.397056 15.6891 0.808831C16.1036 1.22061 16.1036 1.88823 15.6891 2.3L8.75069 9.19117C8.5516 9.38891 8.28156 9.5 8 9.5C7.71844 9.5 7.44841 9.38891 7.24931 9.19117L0.310946 2.3C-0.103649 1.88823 -0.103649 1.22061 0.310946 0.808831Z"
+    />
+  </svg>
+)
+
+const ControlGroupFC: React.FC<ControlGroupProps> = props => {
+  const {height, controlAppearance, groupAppearance} = useDefaults(
+    'controlGroup',
+    props,
+    {
+      height: 40,
+      controlAppearance: 'default' as ControlAppearanceType,
+      groupAppearance: 'button' as ControlGroupAppearanceType
+    }
+  )
+  const {
+    items,
+    onChange,
+    value: groupValue,
+    layout,
+    textAlign,
+    disabled,
+    invalid,
+    visibleCount
+  } = props
+
+  const [controlsNumber, setControlsNumber] = React.useState(0)
+  const [isOpen, setIsOpen] = React.useState(false)
+  const haveMoreButton =
+    controlAppearance === 'outline' && groupAppearance === 'button'
+  const wrapperNode = React.useRef<HTMLDivElement>(null)
+  const moreButtonNode = React.useRef<HTMLButtonElement>(null)
+
+  React.useEffect(() => {
+    setControlsNumber(visibleCount || 0)
+  }, [])
```

Comment:
You can replace this with:

```tsx
const [controlsNumber, setControlsNumber] = React.useState(() => visibleCount || 0)
```

If you pass callback to useState, it will run it only at first render - it has the same effect as using useEffect with [] as dependency.

Reply:
fixed